### PR TITLE
Fix #4737 #4929 #5284 #5539 #5541 #5788 SORTA plugin

### LINKED ISSUE #4737: As a user, if I try to view my SORTA results I get an error in the console: molgenis.StandardModal is not a constructor
## Reproduce
- Use the files and ontology from [here](https://drive.google.com/drive/u/0/folders/0B7qHi41THTs_TmFjSlFhcTlTZkk)
- Import them, configure a user to use SORTA service and Ontology_Ontology
- Create a new task
- Wait till done, then go to result
## Expected

I see my results table
## Observed

I do not have a table
screen shot 2016-04-05 at 13 49 02
screen shot 2016-04-05 at 13 48 57
## Note

As admin, I get the same error in the console, but i DO have a results table

### PR DESCRIPTION
**Pull request by dennishendriksen**
Status:
- [x] Test cases in test plan can be completed successfully for a superuser (Sido)
- [x] execute test plan for a non-superuser
- (n.a.) write unit tests for SortaServiceController
- (n.a.) add unit tests to SortaServiceImplTest
- [x] check impact of updating OntologyTermHitMetaData.java (other code + backwards compatibility)
- [x] list which existing issues are fixed by this PR
- [x] produce ```/search``` issue by hitting code path containing ```new DynamicEntity(null);``` and fix it (removed)
- [x] create technical debt issues in GitHub for remaining FIXMEs and TODOs
- [x] build project with all tests

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- (n.a.) User documentation updated
- (n.a.) (If you have changed REST API interface) view-swagger.ftl updated
- (n.a.) Test plan template updated
- [x] Clean commits

### CODE REVIEW

File: molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/sorta/controller/SortaControllerIT.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+													  .
+															  header(TOKEN_HEADER, getAdminToken())
+													  .with(csrf())
+													  .contentType(APPLICATION_JSON_VALUE)
+													  .accept(APPLICATION_JSON_VALUE)).andExpect(status().isOk());
+	}
+
+	@Test(groups = "withData", dependsOnMethods = "testUploadFile")
+	@WithMockUser(username = SUPERUSER_NAME, roles = SUPERUSER_ROLE)
+	public void testMathEntity() throws Exception
```
Comment (tommydeboer) [inline molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/sorta/controller/SortaControllerIT.java]: typo: testMat**c**hEntity

File: molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/sorta/controller/SortaITUtils.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+			}
+			catch (Exception err)
+			{
+				LOG.error("Error occurred during specific before method. ", err);
+			}
+		});
+	}
+
+	public void addUserIfExists(String userName)
```
Comment (tommydeboer) [inline molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/sorta/controller/SortaITUtils.java]: Shouldn't this be ```addUserUnlessExists()```?

File: molgenis-ontology/src/main/java/org/molgenis/ontology/sorta/service/impl/SortaServiceImpl.java [hunk 3]
```diff
@@ -210,7 +211,8 @@ private void lexicalMatchOntologyTerms(String ontologyIri, Entity inputEntity, E
 		});
 	}
 
-	Entity addLexicalScoreToMatchedEntity(Entity inputEntity, Entity ontologyTerm, String ontologyIri)
+	Entity addLexicalScoreToMatchedEntity(Entity inputEntity, Entity ontologyTerm,
+			String ontologyIri) // TODO Change 'Entity ontologyTerm' to 'OntologyTerm ontologyTerm'
```
[the remark this answers is not in the document]
Comment (sidohaakma) [reply molgenis-ontology/src/main/java/org/molgenis/ontology/sorta/service/impl/SortaServiceImpl.java]: Issue #6873

**Comment by sidohaakma** [comment]
TODO: check impact of updating OntologyTermHitMetaData.java (other code + backwards compatibility)

Check if this was a breaking change. Is not relevant anymore because of other breaking changes that force 6.0.0 release of MOLGENIS.

File: molgenis-ontology/src/main/java/org/molgenis/ontology/sorta/service/impl/SortaServiceImpl.java [hunk 4]
```diff
@@ -199,7 +199,8 @@ private void lexicalMatchOntologyTerms(String ontologyIri, Entity inputEntity, E
 		Stream<Entity> lexicalMatchedOntologyTermEntities = dataService.findAll(ONTOLOGY_TERM,
 				new QueryImpl<>(finalQueryRules).pageSize(pageSize))
 																	   .map(ontologyTerm -> addLexicalScoreToMatchedEntity(
-																			   inputEntity, ontologyTerm, ontologyIri));
+																			   inputEntity, ontologyTerm,
+																			   ontologyIri)); // TODO use findAll(ONTOLOGY_TERM, ..., OntologyTerm.class)
```
[the remark this answers is not in the document]
Comment (dennishendriksen) [reply molgenis-ontology/src/main/java/org/molgenis/ontology/sorta/service/impl/SortaServiceImpl.java]: See https://github.com/molgenis/molgenis/issues/6873

**Pull request merged**